dev/tests/js/jasmine/tests/app/code/Magento/Theme/view/frontend/web/js/cookie-status.test.js

```diff
@@ -0,0 +1,39 @@
+
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+define([
+    'jquery',
+    'cookieStatus'
+], function ($, Cookie) {
+    'use strict';
+
+    describe('Magento_Theme/js/cookie-status', function () {
+        var widget,
+            htmlContainer = '<div id="cookie-status" style="display: none"></div>',
+            navigator;
+
+        beforeEach(function () {
+            $(document.body).append(htmlContainer);
+            widget = new Cookie();
+            navigator = window.navigator;
+        });
+
+        afterEach(function () {
+            window.navigator = navigator;
+        });
+
+        it('verify cookie-status initialization', function () {
```

Comment:
I would change that to `it('shows the modal when cookies are not supported'` and then add a test that modal is not added when cookies are supported.